test/cypress/integration/group-member-removal.spec.js

```diff
@@ -0,0 +1,188 @@
+      cy.getByDT('submitBtn').click()
+    })
+    cy.getByDT('closeModal').should('not.exist')
+    assertMembersCount(1)
+  }
+
+  // ------- Remove member that has only 1 group.
+
+  it('user1 creates a groupA', () => {
+    cy.visit('/')
+    cy.giSignup(`user1-${userId}`, { bypassUI: true })
+
+    cy.giCreateGroup(groupNameA, { bypassUI: true })
+
+    cy.giGetInvitationAnyone().then(url => {
+      invitationLinks.anyone_groupA = url
+    })
+
+    cy.giLogout()
+  })
+
+  it('user2 joins groupA', () => {
+    cy.giAcceptGroupInvite(invitationLinks.anyone_groupA, {
+      username: `user2-${userId}`,
+      groupName: groupNameA,
+      bypassUI: true
+    })
+  })
+
+  it('user1 removes user2 from groupA', () => {
+    cy.giLogin(`user1-${userId}`)
+
+    openRemoveMemberModal('user2', 2)
+    removeMemberNow('user2')
+    cy.giLogout()
+
+    // verify user2 (removed) has no group now.
+    cy.giLogin(`user2-${userId}`) // [*note_1*]
+    cy.getByDT('welcomeHomeLoggedIn').should('contain', 'Let’s get this party started')
+  })
+
+  it('user2 rejoins groupA', () => {
+    // Wait for contracts to sync before visiting invitationLink. TODO avoid .wait().
+    cy.wait(500); // eslint-disable-line
+    cy.giAcceptGroupInvite(invitationLinks.anyone_groupA, {
+      username: `user2-${userId}`,
+      groupName: groupNameA,
+      isLoggedIn: true,
+      shouldLogoutAfter: false
+      // TODO not working. error: [CRITICAL ERROR] NOT EXPECTING EVENT!'
+      // bypassUI: true,
+    })
+    assertMembersCount(2)
+  })
+
+  // ------- Remove member that has a second group.
+
+  it('user2 creates a groupB', () => {
+    cy.giCreateGroup(groupNameB, { bypassUI: true })
+
+    cy.giGetInvitationAnyone().then(url => {
+      invitationLinks.anyone_groupB = url
+    })
+  })
+
+  it('user1 joins groupB - has now 2 groups', () => {
+    cy.giSwitchUser(`user1-${userId}`)
+
+    cy.giAcceptGroupInvite(invitationLinks.anyone_groupB, {
+      username: `user1-${userId}`,
+      groupName: groupNameB,
+      isLoggedIn: true,
+      bypassUI: true
+    })
+  })
+
+  it('user2 removes user1 from groupB - user1 is left with groupA.', () => {
+    cy.giLogin(`user2-${userId}`)
+
+    openRemoveMemberModal('user1', 2)
+    removeMemberNow('user1')
+    cy.giLogout()
+
+    // verify user1 (removed) still has group1.
+    cy.giLogin(`user1-${userId}`) // [*note_1*]
+    cy.getByDT('groupName').should('contain', groupNameA)
+
+    cy.giLogout()
+  })
+
+  // ------- Propose to remove member from group.
+
+  it('userBot joins groupA', () => {
+    cy.giAcceptGroupInvite(invitationLinks.anyone_groupA, {
+      username: `userBot-${userId}`,
+      groupName: groupNameB,
+      bypassUI: true
+    })
+  })
+
+  it('user1 proposes to remove userBot', () => {
+    cy.giLogin(`user1-${userId}`, { bypassUI: true })
+
+    openRemoveMemberModal('userBot', 3)
```

Comment:
@taoeffect we are having random Cypress fails again where some data is not synced at time.
In [this run spec](https://dashboard.cypress.io/projects/q6whky/runs/299/specs) go see the video. User1 logins but doesn't see userBot on the group members' list, although userBot successfully joined the group on the previous test.

The `data-sync` we added to #app during the hackathon doesn't seem to be working correctly. Any suggestions to fix this?

Reply:
@sandrina-p can you find a way to get the browser console log output from that session? It might be something else..

Reply:
I'll research about it tomorrow!